torch/csrc/lazy/backend/backend_device.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <ostream>
+#include <memory>
+#include <string>
+
+#include <c10/macros/Export.h>
+#include <c10/util/Deprecated.h>
+
+namespace torch {
+namespace lazy {
+
+// Backend should extend it and define their own supported hardware types.
+struct TORCH_API BackendDeviceType {
+  int8_t type {0};
+
+  virtual ~BackendDeviceType() = default;
+  virtual std::string toString() const { return "Unknown"; }
```

Comment:
did you consider tradeoffs of making this a pure-interface class vs having implementation that returns 'Unknown'?

Reply:
It's probably just a policy. Do we want to enforce all backends to implement this interface? I think backends should implement this but I don't know if we want to force that. Things can still work even if they don't, for instance, TS backend. Just for UAPI that query the actual hardware type won't work. So, I'm leaning towards to recommended but not required.

Reply:
yea that makes sense.  I am fine with this.